src/pages/MarketDetails/Card/index.tsx

```diff
@@ -1,19 +1,61 @@
 /** @jsxImportSource @emotion/react */
 import React from 'react';
+import Typography from '@mui/material/Typography';
 import Paper from '@mui/material/Paper';
 
+import { IStat } from '../types';
 import { useStyles } from './styles';
 
+export interface ILegend {
+  label: string;
+  color: string;
+}
+
 export interface ICardProps {
   title: string;
+  legends?: ILegend[];
+  stats?: IStat[];
+  className?: string;
 }
 
-const Card: React.FC<ICardProps> = ({ children, title }) => {
+const Card: React.FC<ICardProps> = ({ children, title, legends = [], stats = [], className }) => {
   const styles = useStyles();
 
   return (
-    <Paper>
-      <h4 css={styles.title}>{title}</h4>
+    <Paper className={className}>
+      <div css={styles.header}>
+        <h4 css={styles.title}>{title}</h4>
+
+        {legends.length > 0 && (
+          <div css={styles.row}>
+            {legends.map(legend => (
+              <div css={styles.legend} key={`card-${title}-legend-${legend.label}`}>
+                <div css={styles.getLegendColorIndicator({ color: legend.color })} />
+
+                <Typography css={styles.legendLabel} variant="small2">
+                  {legend.label}
+                </Typography>
+              </div>
+            ))}
+          </div>
+        )}
+      </div>
+
+      {stats.length > 0 && (
+        <div css={styles.row}>
+          {stats.map(stat => (
+            <div css={styles.stat} key={`card-${title}-legend-${stat.label}`}>
+              <Typography variant="small2" component="div" css={styles.statLabel}>
+                {stat.label}
+              </Typography>
+
+              <Typography variant="h4" component="div" css={styles.statValue}>
```

Comment:
I think there is usually a more descriptive element to use than `div` for displaying text. I usually fallback to span if nothing works well. H4 seems like a possible option here for example

Reply:
In this case an `h4` wouldn't be correct, as it's a value we're displaying not a title. I can use a `span` here if you prefer